shared/stateutil/arrays.go

```diff
@@ -0,0 +1,158 @@
+package stateutil
+
+import (
+	"bytes"
+	"errors"
+	"sync"
+
+	"github.com/minio/highwayhash"
+	"github.com/protolambda/zssz/merkle"
+	"github.com/prysmaticlabs/prysm/shared/bytesutil"
+	"github.com/prysmaticlabs/prysm/shared/hashutil"
+)
+
+var (
+	fastSumHashKey = bytesutil.ToBytes32([]byte("hash_fast_sum64_key"))
```

Comment:
This is the same key has sum64, would it cause collisions? Can we use hashutil to keep everything together?